src/parser.cpp

```diff
@@ -1150,6 +1159,19 @@ void Parser::ParserImpl::loadImport(const ImportSourcePtr &importSource, const M
         attribute = attribute->next();
     }
     XmlNodePtr childNode = node->firstChild();
+
+    if (!childNode) {
```

Comment:
Replace with `if (childNode == nullptr) {`.

Reply:
I'm happy to change it, but I want to understand why?  We have a whole load of `while(something)` statements throughout the XML stuff, and I would have thought that `if(!something)` was the corollary of those?

Reply:
Both are (obviously) valid, equivalent, etc. However, `if (childNode == nullptr) {` is the recommended way by tools like ClangTidy (because it makes the code "clearer"). I am not quite sure why it didn't pick it up here (and elsewhere from what I understand). We will have to check our ClangTidy settings.

Reply:
Just to confirm, `while (something)` should also be replaced with `while (something != nullptr)`, `while (something != 0`, etc.

Reply:
2635f862

Reply:
OK.  I'll make a separate issue then - it's all throughout the code.